libs/layout/src/lib/services/online-service/online.service.ts

```diff
@@ -0,0 +1,54 @@
+import { isPlatformBrowser } from '@angular/common';
+import { Inject, Injectable, OnDestroy, PLATFORM_ID } from '@angular/core';
+import { Observable, Subject, fromEvent, takeUntil, tap } from 'rxjs';
+
+/**
+ * A service that provides the currently online status of the application
+ */
+@Injectable({ providedIn: 'root' })
+export class NgxOnlineService implements OnDestroy {
+	/**
+	 * A subject that emits whenever the application is on or offline
+	 */
+	private readonly onlineSubject: Subject<boolean> = new Subject<boolean>();
+
+	/**
+	 * A subject to handle the destroyed flow
+	 */
+	private readonly onDestroySubject: Subject<void> = new Subject();
+
+	/**
+	 * An observable that emits whenever the application is on or offline
+	 */
+	public readonly online$: Observable<boolean> = this.onlineSubject.asObservable();
+
+	constructor(@Inject(PLATFORM_ID) platformId: string) {
+		// Iben: When we're in the browser, listen to the online and offline status of the application
+		if (isPlatformBrowser(platformId)) {
+			// Iben: Handle the on and offline status of the application
+			fromEvent(window, 'online')
```

Comment:
Not a fan of using the `window` directly here. Maybe create an issue that, when we are able to, we use the `WindowService` here to avoid potential problems.

For now it's okay. Just a to do to keep in mind.

Reply:
Yeah, completely agree; it does use the same logic as the `WindowService` does, so this should work.